api/client/webclient/webconfig.go

```diff
@@ -55,6 +55,12 @@ type WebConfig struct {
 	TunnelPublicAddress string `json:"tunnelPublicAddress,omitempty"`
 	// RecoveryCodesEnabled is a flag that determines if recovery codes are enabled in the cluster.
 	RecoveryCodesEnabled bool `json:"recoveryCodesEnabled,omitempty"`
+
+	UI UIConfig `json:"ui,omitempty"`
+}
+
+type UIConfig struct {
```

Comment:
I would prefer if we didn't have to do all of this boilerplate golang work every time we want to add a config option to the UI. I feel like instead it should read and pass `proxy_service.ui` through to the UI without having to manually build a struct.

Reply:
after this commit, no additional structs would be needed. We need 1 spot to update the yaml file, and one to update the json return that gets spit out of `config.js`

So if we wanted to add something like `customBackgroundColor` or whatever in the future, we'd only add two lines. One for the yaml config, and one for the returned `config.js` file.

Am I misunderstanding your comment?

Reply:
Right sorry you won't have to build any new structs but you'll have to add those two lines. You'll have to know, or investigate, where to add those two lines to get that data from the Teleport config file to the UI. Instead I'd like to see any value that the user puts into that config be passed to the UI so that the only places anyone will have to modify to add a customizable variable is the config file itself and the UI component looking for that value.

Reply:
We'll have to define a field in the `fileconf.go` struct regardless so that the file config marshaler knows it's an accepted field in the yaml file. I don't believe we can have it marshal arbitrary values.

Reply:
Actually it came out quite sick! The line right under this comment is the only place you'll have to touch to add any config to be sent through to the web on the whole backend (docs withstanding of course)

Reply:
I'm not so sure about the dependencies this introduced. Let's discuss on Tuesday.